client/src/payment/index.js

```diff
@@ -0,0 +1,242 @@
+import React from 'react';
+import './payment.css'
+import DetailConfirm from './detailConfirm';
+import DeliveryInfor from './deliveryInfor';
+import PaymentSelect from './paymentSelect';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Stepper from '@material-ui/core/Stepper';
+import Step from '@material-ui/core/Step';
+import StepButton from '@material-ui/core/StepButton';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import Card from '@material-ui/core/Card';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+
+const styles = theme => ({
+    root: {
+        width: '90%',
+    },
+    button: {
+        marginRight: theme.spacing.unit,
+    },
+    backButton: {
+        marginRight: theme.spacing.unit,
+    },
+    completed: {
+        display: 'inline-block',
+    },
+    instructions: {
+        marginTop: theme.spacing.unit,
+        marginBottom: theme.spacing.unit,
+    },
+    card: {
+        maxWidth: 345,
+    },
+    media: {
+        height: 140,
+    },
+});
+
+function getSteps() {
+    return ['Detail Confirmation', 'Delivery Information', 'Payment Select'];
+}
+
+function getStepContent(step) {
+    switch (step) {
+        case 0:
+            return (<DetailConfirm/>);
+        case 1:
+            return (<DeliveryInfor/>);
+        case 2:
+            return (<PaymentSelect/>);
+        default:
+            return 'Unknown step';
+    }
+}
+
+class Payment extends React.Component {
+    state = {
+        activeStep: 0,
+        completed: new Set(),
+        skipped: new Set(),
+    };
+
+    totalSteps = () => {
+        return getSteps().length;
+    };
+
+    isStepOptional = step => {
+        return step === 1;
+    };
+
+    handleNext = () => {
+        let activeStep;
+
+        if (this.isLastStep() && !this.allStepsCompleted()) {
+            // It's the last step, but not all steps have been completed
+            // find the first step that has been completed
+            const steps = getSteps();
+            activeStep = steps.findIndex((step, i) => !this.state.completed.has(i));
+        } else {
+            activeStep = this.state.activeStep + 1;
+        }
+        this.setState({
+            activeStep,
+        });
+    };
+
+    handleBack = () => {
```

Comment:
handleBack = () => {
    const { activeStep } = this.state;
    this.setState({
      activeStep: activeStep - 1,
    });
  };

plz use more deconstructing.